infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/replication/LogReplicationAckReader.java

```diff
@@ -74,13 +80,18 @@ public void setAckedTsAndSyncType(long ackedTs, LogReplicationMetadata.Replicati
         }
     }
 
+    public void startAckReader(LogEntryReader logEntryReader) {
```

Comment:
why do we start this later?

Reply:
because it needs the LogEntryReader which comes from the FSM, which in turn the FSM needs the logReplicationAckReader, so I need to set it after FSM is initialized. Take a look at class LogReplicationSourceManager